xlators/meta/src/meta-helpers.c

```diff
@@ -331,3 +331,36 @@ fixed_dirents_len(struct meta_dirent *dirents)
 
     return i;
 }
+
+int
+file_write_wrapper(xlator_t *this, fd_t *fd, struct iovec *iov, int count,
+                   int (*file_write_str)(xlator_t *this, fd_t *fd, char *str))
+{
+    int i = 0;
+    int j = 0;
+    size_t tlen = 0;
+    char *str = NULL;
+    char *p = NULL;
+    int ret = 0;
+
+    for (i = 0; i < count; i++) {
+        for (j = 0; j < iov[i].iov_len; j++) {
+            if (((char *)iov[i].iov_base)[j] == 0)
+                return -EINVAL;
+        }
```

Comment:
I think it will be more efficient doing this:
```suggestion
        if (memchr(iov[i].iov_base, 0, iov[i].iov_len) != NULL) {
            return -EINVAL;
        }
```

Reply:
applied